src/modm/ui/animation/base.hpp

```diff
@@ -23,6 +23,39 @@ namespace modm
 namespace ui
 {
 
+/// @ingroup modm_ui_animation
+template< typename T >
+class AnimationBase
+{
+public:
+	using TimeType = uint32_t;
+
+	/// stop any running animation and set a value.
+	virtual void
+	setValue(T value) = 0;
+
+	/// @return `true` if animation is currently running,
+	///			`false` if otherwise
+	virtual bool
+	isAnimating() const = 0;
+
+	/// stops any running animation.
+	virtual void
+	stop() = 0;
+
+	/// Animate from the current value to a new value in the specified ms.
+	/// @return `true` if animation started,
+	///			`false` if otherwise
+	virtual bool
+	animateTo(T value, uint32_t time) = 0;
+
+	/// @return	`true` if the value has been changed,
+	///			`false` otherwise
+	virtual bool
+	update()
+	{ return false; }
+};
```

Comment:
This base class should either have a public virtual or protected non-virtual destructor, see http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rc-dtor-virtual.